presto-main/src/main/java/com/facebook/presto/operator/scalar/ArraysOverlapFunction.java

```diff
@@ -0,0 +1,273 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.OperatorDependency;
+import com.facebook.presto.spi.function.ScalarFunction;
+import com.facebook.presto.spi.function.SqlNullable;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.function.TypeParameter;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.facebook.presto.spi.type.Type;
+import it.unimi.dsi.fastutil.ints.AbstractIntComparator;
+import it.unimi.dsi.fastutil.ints.IntArrays;
+import it.unimi.dsi.fastutil.ints.IntComparator;
+
+import java.lang.invoke.MethodHandle;
+import java.util.Arrays;
+
+import static com.facebook.presto.spi.function.OperatorType.LESS_THAN;
+
+@ScalarFunction("arrays_overlap")
+@Description("Returns true if arrays have common elements")
+public final class ArraysOverlapFunction
+{
+    private static final int INITIAL_SIZE = 128;
+
+    private int[] leftPositions = new int[INITIAL_SIZE];
+    private int[] rightPositions = new int[INITIAL_SIZE];
+
+    private long[] arrayBigIntLeft;
+    private long[] arrayBigIntRight;
+    private int[] arrayIntLeft;
+    private int[] arrayIntRight;
+
+    @TypeParameter("E")
+    public ArraysOverlapFunction(@TypeParameter("E") Type elementType) {}
+
+    public ArraysOverlapFunction() {}
+
+    private static IntComparator intBlockCompare(Type type, Block block)
+    {
+        return new AbstractIntComparator()
+        {
+            @Override
+            public int compare(int left, int right)
+            {
+                if (block.isNull(left) && block.isNull(right)) {
+                    return 0;
+                }
+                if (block.isNull(left)) {
+                    return 1;
+                }
+                if (block.isNull(right)) {
+                    return -1;
+                }
+                return type.compareTo(block, left, block, right);
+            }
+        };
+    }
+
+    @SqlNullable
+    @SqlType(StandardTypes.BOOLEAN)
+    public Boolean arraysOverlapInt(
+            @OperatorDependency(operator = LESS_THAN, returnType = StandardTypes.BOOLEAN, argumentTypes = {"integer", "integer"}) MethodHandle lessThanFunction,
+            @SqlType("array(integer)") Block leftArray,
```

Comment:
No, there shouldn't problems like this. I didn't know that AbstractLongType can extraction for me. It reduces code significantly.